src/rgw/rgw_common.cc

```diff
@@ -513,7 +513,7 @@ bool parse_rfc2616(const char *s, struct tm *t)
 bool parse_iso8601(const char *s, struct tm *t, uint32_t *pns, bool extended_format)
 {
   memset(t, 0, sizeof(*t));
-  const char *p;
+  const char *p = nullptr;
```

Comment:
@rzarzynski That's true and the statement `date = d` in `parse_v4_auth_header()` won't be executed. Then why the coverity shows the below report ? I think it is a false positive and thinking of closing this PR. What do you say ?

```
Explicit null dereferenced (FORWARD_NULL)11. var_deref_model: Passing
null pointer d to basic_string_view, which dereferences it.
```